controllers/toolchainstatus/toolchainstatus_controller.go

```diff
@@ -292,14 +301,32 @@ func (r *Reconciler) hostOperatorHandleStatus(reqLogger logr.Logger, toolchainSt
 
 	// check host operator deployment status
 	deploymentConditions := status.GetDeploymentStatusConditions(r.Client, hostOperatorDeploymentName, toolchainStatus.Namespace)
-	err = status.ValidateComponentConditionReady(deploymentConditions...)
-	if err != nil {
-		reqLogger.Error(err, "host operator deployment is not ready")
+	errDeploy = status.ValidateComponentConditionReady(deploymentConditions...)
+	if errDeploy != nil {
+		reqLogger.Error(errDeploy, "host operator deployment is not ready")
 	}
-	// update toolchainStatus
 	operatorStatus.Conditions = deploymentConditions
+	// update toolchainStatus
 	toolchainStatus.Status.HostOperator = operatorStatus
-	return err == nil
+
+	// if we are running in production we also
+	// check that deployed version matches the latest commit from source code repository
+	var errVersionCheck error
+	if isProdEnvironment(r.toolchainConfig) {
+		versionCondition := status.CheckDeployedVersionIsUpToDate(r.GithubClient, hostOperatorRepoName, hostOperatorRepoBranchName, version.Commit)
+		errVersionCheck = status.ValidateComponentConditionReady([]toolchainv1alpha1.Condition{*versionCondition}...)
+		toolchainStatus.Status.HostOperator.Conditions = append(toolchainStatus.Status.HostOperator.Conditions, *versionCondition)
```

Comment:
the `GetDeploymentStatusConditions` function that is called above already contains the `Ready` condition, and we should not have two `Ready` conditions in the same list of conditions.
There are three solutions I can thing of:
1. update the already existing `Ready` condition - when it's already True and the version check fails, then set it to false.
2. move the version check into `GetDeploymentStatusConditions` and incorporate it into the logic there
3. create a new component for the version check

in the case of the 1. and the 2. option, you could also change the logic of ValidateComponentConditionReady, so it would return only the error message in the case the version is out-to-date

Reply:
I've updated the code implementing something similar to option 1), now it adds only one `Ready` condition , please check https://github.com/codeready-toolchain/host-operator/pull/819/commits/d0384c77815b726c068230f89c018fdce41f3c5e